plugins/auth-backend/src/authPlugin.ts

```diff
@@ -19,12 +19,46 @@ import {
   createBackendPlugin,
 } from '@backstage/backend-plugin-api';
 import {
+  authOwnershipResolutionExtensionPoint,
+  AuthOwnershipResolutionExtensionPoint,
+  AuthOwnershipResolver,
   AuthProviderFactory,
+  AuthProviderRegistrationOptions,
+  AuthProvidersExtensionPoint,
   authProvidersExtensionPoint,
 } from '@backstage/plugin-auth-node';
 import { catalogServiceRef } from '@backstage/plugin-catalog-node/alpha';
 import { createRouter } from './service/router';
 
+class AuthProvidersExtensionPointImpl implements AuthProvidersExtensionPoint {
+  #providers = new Map<string, AuthProviderFactory>();
+  registerProvider(options: AuthProviderRegistrationOptions): void {
+    const { providerId, factory } = options;
+    if (this.#providers.has(providerId)) {
+      throw new Error(`Auth provider '${providerId}' was already registered`);
+    }
+    this.#providers.set(providerId, factory);
+  }
+
+  get providers() {
+    return this.#providers;
+  }
+}
+
+class AuthOwnershipResolutonExtensionPointImpl
+  implements AuthOwnershipResolutionExtensionPoint
+{
+  #ownershipResolver?: AuthOwnershipResolver;
+
+  setAuthOwnershipResolver(ownershipResolver: AuthOwnershipResolver): void {
+    this.#ownershipResolver = ownershipResolver;
```

Comment:
Let's throw an error if an ownership resolver is already set at this point, since that would lead to potentially flaky behavior.